pkg/kratos/handlers.go

```diff
@@ -193,6 +199,159 @@ func (a *API) handleKratosError(w http.ResponseWriter, r *http.Request) {
+	}
+
+	flow, cookies, err := a.service.CreateBrowserRecoveryFlow(context.Background(), returnTo, r.Cookies())
+	if err != nil {
+		a.logger.Errorf("Failed to create recovery flow: %v\n", err)
+		http.Error(w, "Failed to create recovery flow", http.StatusInternalServerError)
+		return
+	}
+
+	resp, err := flow.MarshalJSON()
+	if err != nil {
+		a.logger.Errorf("Error when marshalling json: %v\n", err)
+		http.Error(w, "Failed to marshal json", http.StatusInternalServerError)
+		return
+	}
+
+	setCookies(w, cookies)
+	w.WriteHeader(http.StatusOK)
+	w.Write(resp)
+}
+
+func (a *API) handleGetSettingsFlow(w http.ResponseWriter, r *http.Request) {
+	q := r.URL.Query()
+
+	flow, cookies, err := a.service.GetSettingsFlow(context.Background(), q.Get("id"), r.Cookies())
+	if err != nil {
+		a.logger.Errorf("Error when getting settings flow: %v\n", err)
+		http.Error(w, "Failed to get settings flow", http.StatusInternalServerError)
+		return
+	}
+
+	resp, err := flow.MarshalJSON()
+	if err != nil {
+		a.logger.Errorf("Error when marshalling json: %v\n", err)
+		http.Error(w, "Failed to parse settings flow", http.StatusInternalServerError)
+		return
+	}
+	setCookies(w, cookies)
+	w.WriteHeader(http.StatusOK)
+	w.Write(resp)
+}
+
+func (a *API) handleUpdateSettingsFlow(w http.ResponseWriter, r *http.Request) {
+	q := r.URL.Query()
+	flowId := q.Get("flow")
+
+	body, err := a.service.ParseSettingsFlowMethodBody(r)
+	if err != nil {
+		a.logger.Errorf("Error when parsing request body: %v\n", err)
+		http.Error(w, "Failed to parse settings flow", http.StatusInternalServerError)
+		return
+	}
+
+	flow, cookies, err := a.service.UpdateSettingsFlow(context.Background(), flowId, *body, r.Cookies())
+	if err != nil {
+		a.logger.Errorf("Error when updating settings flow: %v\n", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	resp, err := json.Marshal(flow)
+	if err != nil {
+		a.logger.Errorf("Error when marshalling json: %v\n", err)
+		http.Error(w, "Failed to parse settings flow", http.StatusInternalServerError)
+		return
+	}
+	setCookies(w, cookies)
+	w.WriteHeader(http.StatusOK)
+	w.Write(resp)
+}
+
+func (a *API) handleCreateSettingsFlow(w http.ResponseWriter, r *http.Request) {
+	returnTo, err := url.JoinPath(a.baseURL, "/ui/reset_complete")
+	if err != nil {
+		a.logger.Errorf("Failed to construct returnTo URL: ", err)
+		http.Error(w, "Failed to construct returnTo URL", http.StatusBadRequest)
+	}
+
+	flow, cookies, err := a.service.CreateBrowserSettingsFlow(context.Background(), returnTo, r.Cookies())
+	if err != nil {
+		a.logger.Errorf("Failed to create settings flow: %v", err)
+		http.Error(w, "Failed to create settings flow", http.StatusInternalServerError)
+		return
+	}
+
+	resp, err := flow.MarshalJSON()
+	if err != nil {
+		a.logger.Errorf("Error when marshalling json: %v\n", err)
+		http.Error(w, "Failed to marshal json", http.StatusInternalServerError)
+		return
+	}
+	setCookies(w, cookies)
```

Comment:
you might want to drop https://github.com/canonical/identity-platform-login-ui/pull/245/files#diff-12151c8a9259cc64c5b82a9a51ed3eb6367a4850988ecd45cb323b4e086c708cR372

Reply:
This handler is still used in the login flow (see [example](https://github.com/canonical/identity-platform-login-ui/blob/65e63207deb03770e1b17a5278d585de1ea4b458/pkg/kratos/service.go#L52)). Can I refactor it in another PR since I'm addressing recovery and settings in this one?
Some other comments you made or `ioutil` deprecation are also relevant for the login flow

Reply:
missed that, yep that's fair, goog to go then